CHANGES.md

```diff
@@ -1,6 +1,362 @@
+* renamed `FileSystem.isLink()` to `FileSystem.isSymLink()`  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/FileSystem.html#FileSystem.isSymlink)
+* renamed several 'FileSystem' module symbols to follow naming conventions
+* renamed the `Timer` type to `stopwatch` in the 'Time' module  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/Time.html#Time.stopwatch)
+* renamed several 'Time' module symbols to follow naming conventions
+* renamed the `_file` type to `c_FILE` in the 'CTypes' module  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/CTypes.html#CTypes.c_FILE)
+* renamed `sourceVersion` type to `versionValue` in the 'Version' module  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/Version.html#Version.versionValue)
+
+Deprecated / Unstable / Removed Library Features
+------------------------------------------------
+* deprecated 'file.localesForRegion()' in the 'IO' module  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/IO.html#IO.file.localesForRegion)
+* deprecated `unicodeSupported` from 'IO' because it always returned `true`  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/IO.html#IO.unicodeSupported)
+* deprecated `string`/`bytes` casts to `regex` in favor of `Regex.compile()`  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/Regex.html#Regex.compile)
+* deprecated `copyMode()`, `getMode()`, and `chmod()` from 'FileSystem'
+* deprecated `createVersion()` in favor of `new versionValue()`  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/Version.html#Version.createVersion)
+* removed `"version "` prefix when casting a `sourceVersion` to string  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/Version.html#Version.sourceVersion)
+* deprecated the 'SysBasic' module including `EEOF`, `ESHORT`, `EFORMAT`
+* marked 'GPU'/'GPUDiagnostics' modules as unstable (due to still being new))
+* marked `iostringstyle`, `iostringformat` as unstable in the 'IO' module
+* removed the deprecated 'SysError' module and some related deprecated symbols
+* removed the deprecated 'DateTime' module
+* removed deprecated support for `start`/`end` arguments from 'IO' routines
+* removed `tryGetPath()`, `unlink()`, and `getchunk()` from 'IO'
+* removed deprecated `iohint` type from the 'IO' module
+* removed deprecated formals in channel methods `.search()`, `.match()`, etc.
+* removed support for deprecated formals in `regex.compile()`
+* removed deprecated class `BadRegexpError`
+* removed deprecated fields `.offset`, `.size` in `regexMatch`
+* removed deprecated `.indexOf()`, `match()` methods on `list`
+* removed deprecated `map` operators
+
+Standard Library Modules
+------------------------
+* added `readAll()` methods to read the entire contents of a `fileReader`  
+  (see https://chapel-lang.org/docs/1.29/modules/standard/IO.html#IO.channel.readAll)
+* added six new overloads of `read`|`writeBinary()` for a variety of types  
```

Comment:
I have a hard time parsing this type of shortening. Not sure if it's read/writeBinary or readBinary/writeBinary. I'm used to a style like `{read,write}Binary` from bash, but don't think that's a syntax we use here. Maybe:

```
* added overloads of `readBinary()` and `writeBinary()` for a variety of types
```